### PR TITLE
Update RV32 and RV64 architectures

### PR DESCRIPTION
**Pull request by wysiwyng**
This adds updates to the RV32 and RV64 architecture models, namely:

- CoreDSL 2 generated trap entry and supporting ETISS fixes
- CoreDSL 2 generated semihosting capabilities and supporting ETISS core additions
- A new RV64IMACFD + Zicsr + Zifencei compatible model that passes all U-Mode and almost all S- and M-Mode instruction tests
- Generated architecture model code makes better use of ETISS' `CodeParts` feature, allowing plugins to better insert their own JIT-Code

Interrupt support of these architecture models is currently untested and probably not implemented correctly. Adding interrupt support would be a matter of extending the trap entry function in CoreDSL to handle incoming interrupts, as well as extending ETISS to generate a corresponding call to said trap entry function.

RV64M needs 128 bit arithmetic, this is not supported per se in ETISS, see also #107. This patch uses the GCC `__[u]int128_t` types for limited support until a better solution is found. Because of this, RV64M only works when using GCC or LLVM as JIT engine, as TCC has no 128-bit arithmetic support.

### CODE REVIEW

**Review by JoGei — COMMENTED** [review]
As discussed in #116 I would agree with the position that "exiting on loop to self" is not a good idea.

The `wfi` is not defined to must set the hart to sleep, i.e., is implementation specific.
I also do not think it is a bad take to force the application code to decide when to exit the simulation, e.g., by semihosting + `ecall`s. 
However, most importantly the self-loop-exit behavior is not described in any standard and will/might lead to problems in future scenarios.

**Comment by JoGei** [comment]
> As discussed in #116 I would agree with the position that "exiting on loop to self" is not a good idea.
> 
> The `wfi` is not defined to must set the hart to sleep, i.e., is implementation specific. I also do not think it is a bad take to force the application code to decide when to exit the simulation, e.g., by semihosting + `ecall`s. However, most importantly the self-loop-exit behavior is not described in any standard and will/might lead to problems in future scenarios.

Just saw that in the commit https://github.com/tum-ei-eda/etiss/pull/127/commits/44f695f25d9a11f8ea34cd35e8985da25c01ffc3 the "exit-on-loop-to-self" was made optional with a configuration setting.
All good. :+1:

**Comment by wysiwyng** [comment]
Converted to draft until interrupt support is ready

**Comment by wysiwyng** [comment]
New updates: 
* Interrupt handling:
    * Fully functional
    * Fully CoreDSL-generated
    * Added `InterruptEnable` class to query a global interrupt enable flag in ETISS' `InterruptHandler` plugin
* CMake:
    * Added code to link architecture plugins against JIT libraries to use functions from those in non-JIT code
* RV32/64 Architecture plugins:
    * fixed various remaining issues, `M/SSTATUS` CSRs, floating point instructions
    * `riscv-tests` related to MMU, CSR and breakpoints fail, all others pass
* General:
    * Prepare various bits for new MMU integration

File: ArchImpl/RV64IMACFD/RV64IMACFDArchSpecificImp.h [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	virtual uint64_t _read() const {
+		return (uint64_t) *((RV64IMACFD*)parent_.structure_)->X[gprid_];
+	}
+
+	virtual void _write(uint64_t val) {
+		etiss::log(etiss::VERBOSE, "write to ETISS cpu state", name_, val);
+		*((RV64IMACFD*)parent_.structure_)->X[gprid_] = (etiss_uint64) val;
+	}
+};
+
```
[the remark this answers is not in the document]
Comment (wysiwyng) [reply ArchImpl/RV64IMACFD/RV64IMACFDArchSpecificImp.h:57]: Will be added later once these can be generated properly by M2-ISA-R

[the remark this answers is not in the document]
Comment (JoGei) [reply ArchImpl/RV64IMACFD/RV64IMACFDArchSpecificImp.h:57]: I have added the needed Fields manually (https://github.com/JoGei/etiss/commit/12db1142d6c010f8be0014697f121f13db97369f) to test VirtualStruct with this architecture update.

**Pull request merged**